### DISCUSSION TITLE
How to detect HID in WSL2? (node-hid)

### DISCUSSION THREAD

**Discussion by do-kevin** [microsoft/WSL#9594] [Q&A]
Hey folks,

So I'm learning how to access HID in general and I wanted to do it in WSL2.
I've followed [https://devblogs.microsoft.com/commandline/connecting-usb-devices-to-wsl](https://devblogs.microsoft.com/commandline/connecting-usb-devices-to-wsl) to set up.

I'm trying to detect my controller. The bus. vendor id, and whatever of my device do show up when I do `lsusb`. So I am using `node-hid` in my NodeJS app.

But when I run my code like this:
```
const hid = require(node-hid);
const devices = hid.devices();
console.log("result: ", devices);
```

I get an empty array. Is there some additional step I am missing, a driver, a missing package I need to install in Linux?

**Comment by Dheeraj-Bhaskaruni** [comment]
same question any luck finding the answer?

**Comment by dnmeid** [comment]
The older kernels do not support higher level hid access, only generic usb.
HIDRAW just has been added recently with kernel 5.15.150.1 https://github.com/microsoft/WSL2-Linux-Kernel/releases/tag/linux-msft-wsl-5.15.150.1
